src/dbnode/persist/fs/seek_manager.go

```diff
@@ -933,7 +957,10 @@ func (m *seekerManager) openCloseLoop() {
 			byTime.RLock()
 			for blockStartNano := range byTime.seekers {
 				blockStart := blockStartNano.ToTime()
-				if blockStart.Before(earliestSeekableBlockStart) {
+				if blockStart.Before(earliestSeekableBlockStart) ||
+					// Close seekers for shards that are no longer available. This
+					// ensure that seekers are eventually consistent w/ shard state.
+					!m.isShardAvailableWithLock(uint32(shard)) {
```

Comment:
Actually we don't want to remove the shard until it's not in the shardset at all.

Leaving shards could be restored to Available if the join gets aborted, and the Initializing shard won't serve reads until it's bootstrapped anyway.